format_spec.md

```diff
@@ -313,8 +313,23 @@ This field MUST be either an Array or String. If this field is an array, it MUST
 
 ##### `src` (optional)
 
-If not all information is listed inside a dependency object and it's metadata is not hosted on the same host which is serving metadata for the parent project, the `src` field may be used. This field MUST contain a URL. This URL MUST serve MODIP Format-compliant metadata about the required dependency. It MUST include the `versions` field.  
-    
+This specifies where information about this dependency can be acquired, but it's not always required. If the metadata of the parent project is being served from a host, then this field is not required and clients can lookup the ID at the host. For example, if I request metadata about a project from Modrinth and there is a dependency, "my-dependency", then I should call Modrinth's API in order to find out more about that dependency.
+
+But it some cases, that's either not possible or impractical, which is where the `src` field comes in. This field MUST contain either a String containing a URL that links to full metadata about the dependency, or it can be an object which fully complies with the MODIP project spec.
```

Comment:
Just curious - what's the use case for inline dependencies? Easier sharing of MODIP files?
How can they be deduplicated if more than one project has the same inline dependency,?

Reply:
I was writing pack-conversion-tool to convert Packwiz to MODIP, and mods in a modpack are stored as a dependency with MODIP. I needed to include information about the mods in the pack, but keep it local and on-disk as none of them were available remotely in MODIP format. That’s where I got the idea from. It’s good for representing a mod that’s not served remotely anywhere in MODIP format.

I could have also done a “one file per mod” approach, and have the relative file path to the mods metadata as the `src` value. It’s still an option, but it’s a bit more complicated than necessary.

As for deduplication - I’m not sure. The problem also exists with regular, non-inline dependencies too so we’ll have to figure that out.